### PR TITLE
Move a couple of functions out of line.

### PR DESCRIPTION
**Pull request by bangerth**
/rebuild

### CODE REVIEW

**Review by marcfehling — APPROVED** [review]
Looks good to me.

Do you want to keep the function definitions here, or move them to `source/sundials/arkode.cc`?

File: include/deal.II/sundials/arkode.h [hunk 1]
```diff
@@ -1349,6 +1308,70 @@ namespace SUNDIALS
                  << " returned a negative error code: " << arg1
                  << ". Please consult SUNDIALS manual.");
 
+
+  template <typename VectorType = Vector<double>>
```
Comment (marcfehling) [inline include/deal.II/sundials/arkode.h:1312]:
```suggestion
  template <typename VectorType>
```

File: include/deal.II/sundials/arkode.h [hunk 2]
```diff
@@ -1349,6 +1308,70 @@ namespace SUNDIALS
...
+    , implicit_function_is_linear(implicit_function_is_linear)
+    , implicit_function_is_time_independent(
+        implicit_function_is_time_independent)
+    , mass_is_time_independent(mass_is_time_independent)
+    , anderson_acceleration_subspace(anderson_acceleration_subspace)
+  {}
+
+
+
+  template <typename VectorType = Vector<double>>
```
Comment (marcfehling) [inline include/deal.II/sundials/arkode.h:1348]:
```suggestion
  template <typename VectorTyp>
```

**Review by marcfehling — COMMENTED** [review]
```
In file included from /jenkins/workspace/dealii_PR-13436/source/sundials/arkode.cc:19:0:
/jenkins/workspace/dealii_PR-13436/include/deal.II/sundials/arkode.h:1329:36: error: default argument for template parameter for class enclosing ‘dealii::SUNDIALS::ARKode<VectorType>::AdditionalData::AdditionalData(double, double, double, double, double, unsigned int, unsigned int, bool, bool, bool, int, double, double)’
     const double relative_tolerance)
                                    ^
/jenkins/workspace/dealii_PR-13436/include/deal.II/sundials/arkode.h:1350:75: error: default argument for template parameter for class enclosing ‘void dealii::SUNDIALS::ARKode<VectorType>::AdditionalData::add_parameters(dealii::ParameterHandler&)’
   ARKode<VectorType>::AdditionalData::add_parameters(ParameterHandler &prm)
                                                                           ^
```

**Review by masterleinad — APPROVED** [review]

**Comment by bangerth** [comment]
I was all excited by the fact that Eclipse did this refactoring for me. Too bad it didn't get it right :-(

**Review by kronbichler — APPROVED** [review]

**Pull request merged**